packages/react/src/firestore/useGetCountFromServerQuery.ts

```diff
@@ -0,0 +1,24 @@
+import { useQuery, type UseQueryOptions } from "@tanstack/react-query";
+import {
+  getCountFromServer,
+  type FirestoreError,
+  type Query,
+} from "firebase/firestore";
+
+type FirestoreUseQueryOptions<TData = unknown, TError = Error> = Omit<
+  UseQueryOptions<TData, TError>,
+  "queryFn"
+>;
+
+export function useGetCountFromServerQuery(
+  query: Query,
+  options: FirestoreUseQueryOptions<number, FirestoreError>
+) {
+  return useQuery<number, FirestoreError>({
+    ...options,
+    queryFn: async () => {
+      const snapshot = await getCountFromServer(query);
+      return snapshot?.data().count;
```

Comment:
Any idea why would the snapshot here ever be nullish? From the API docs it should always return a snapshot: https://firebase.google.com/docs/reference/js/firestore_.md#getcountfromserver_4e56953

Reply:
true, ive actually removed the optional chaining